Fix es_ui_shared eslint violations for useRequest hook

cjcenizal opened the pull request:
Replaces https://github.com/elastic/kibana/pull/59626

---

Partially addresses https://github.com/elastic/kibana/issues/49554

Closes https://github.com/elastic/kibana/issues/49572 (originally fixed as part of the NP migration)
Closes https://github.com/elastic/kibana/issues/49562 (originally fixed as part of the NP migration)
Closes https://github.com/elastic/kibana/issues/42225 (flaky test)
Closes https://github.com/elastic/kibana/issues/42563 (flaky test)
Closes https://github.com/elastic/kibana/issues/42562 (flaky test)
Closes https://github.com/elastic/kibana/issues/42561 (flaky test)

This PR:
* Removes the unused original `request.ts` helper and replaces it with the NP-ready version
* ~~Adds clean-up logic for when the component using this helper is unmounted~~ (This was reverted)
* Unskips the tests and rewrites them to run synchronously using fake timers.
* Converts the tests to TypeScript.
* Relaxes the `SendRequestResponse` type definition to type `error` as `any` instead of `Error`, since we weren't actually fulfilling that type.

To test:
* Verify that requests behave as expected in Index Management, Watcher, Snapshot Restore, Ingest Manager, and Transforms

## Errors

I tested error-handling using data streams.

### ES errors

#### 405
![image](https://user-images.githubusercontent.com/1238659/92198325-f08b8780-ee28-11ea-80c8-0473525abacf.png)

#### 400
![image](https://user-images.githubusercontent.com/1238659/92198332-f3867800-ee28-11ea-8292-e012e0df79d7.png)

#### `caused_by` chain

I was concerned that the change in the way `sendRequest` surfaced errors would change the way `caused_by` chains were being surfaced in the UI, but it looks like the two apps that use this module and surface `caused_by` chains actually return them as part of the success response, not the error response. These apps are Snapshot Restore (deletRepositories endpoint) and Index Management (deleteDataStreams and deleteIndexTemplates endpoints).

![image](https://user-images.githubusercontent.com/1238659/92331774-db337a80-f02d-11ea-856f-a1fc3d919c55.png)
![image](https://user-images.githubusercontent.com/1238659/92331805-159d1780-f02e-11ea-98c5-db9aa2d75d4d.png)

### Non-ES errors

#### 500
![image](https://user-images.githubusercontent.com/1238659/92198343-faad8600-ee28-11ea-923f-2eced1837c18.png)

#### 500
![image](https://user-images.githubusercontent.com/1238659/92198349-fe410d00-ee28-11ea-804f-23fefb329223.png)

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63736/)
* Commit: 62ac4ea0978f1317186c547181d50525fb0f9359

### Build metrics
#### page load bundle size
| id | [value](https://github.com/elastic/kibana/commit/62ac4ea0978f1317186c547181d50525fb0f9359) | diff | [baseline](https://github.com/elastic/kibana/commit/e9ec039e8e60811eced7fdc95183a64943ad3cfd) |
| --- | --- | --- | --- |
| esUiShared | 1001.0KB | -704.0B | 1001.7KB |
| indexManagement | 255.5KB | -7.0B | 255.5KB |
| total | - | -711.0B | - |

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63736/","number":"63736","commit":"62ac4ea0978f1317186c547181d50525fb0f9359"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/64562/)
* Commit: 9785d3d06eeec86a83738b85276dd65cb1810102
* This comment will update when the build is complete

### Failed CI Steps
* [Execute kibana-intake](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/64562/execution/node/312/log)

### History
* :green_heart: [Build #63736](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63736/) succeeded 62ac4ea0978f1317186c547181d50525fb0f9359

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63736/","number":"63736","commit":"62ac4ea0978f1317186c547181d50525fb0f9359"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/64562/","number":"64562","commit":"9785d3d06eeec86a83738b85276dd65cb1810102"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65001/)
* Commit: 528c28465051eb9a4469794d0b50bb3c696a1132
* This comment will update when the build is complete

### Failed CI Steps
* [Execute kibana-intake](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65001/execution/node/194/log)

### History
* :broken_heart: [Build #64562](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/64562/) failed 9785d3d06eeec86a83738b85276dd65cb1810102
* :green_heart: [Build #63736](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63736/) succeeded 62ac4ea0978f1317186c547181d50525fb0f9359

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63736/","number":"63736","commit":"62ac4ea0978f1317186c547181d50525fb0f9359"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/64562/","number":"64562","commit":"9785d3d06eeec86a83738b85276dd65cb1810102"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65001/","number":"65001","commit":"528c28465051eb9a4469794d0b50bb3c696a1132"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65421/)
* Commit: 446e94b38660616f2cc66d5019cf03e2f26b06f2
* This comment will update when the build is complete

### Failed CI Steps
* [Execute kibana-intake](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65421/execution/node/308/log)

### History
* :broken_heart: [Build #65001](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65001/) failed 528c28465051eb9a4469794d0b50bb3c696a1132
* :broken_heart: [Build #64562](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/64562/) failed 9785d3d06eeec86a83738b85276dd65cb1810102
* :green_heart: [Build #63736](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63736/) succeeded 62ac4ea0978f1317186c547181d50525fb0f9359

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63736/","number":"63736","commit":"62ac4ea0978f1317186c547181d50525fb0f9359"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/64562/","number":"64562","commit":"9785d3d06eeec86a83738b85276dd65cb1810102"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65001/","number":"65001","commit":"528c28465051eb9a4469794d0b50bb3c696a1132"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65421/","number":"65421","commit":"446e94b38660616f2cc66d5019cf03e2f26b06f2"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65572/)
* Commit: 0975904b04fe069eb8e254e7976fa5dbcd224739
* This comment will update when the build is complete

### Failed CI Steps
* [Execute kibana-ciGroup1](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65572/execution/node/372/log)
* [Execute kibana-ciGroup1](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65572/execution/node/438/log)

### History
* :broken_heart: [Build #65421](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65421/) failed 446e94b38660616f2cc66d5019cf03e2f26b06f2
* :broken_heart: [Build #65001](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65001/) failed 528c28465051eb9a4469794d0b50bb3c696a1132
* :broken_heart: [Build #64562](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/64562/) failed 9785d3d06eeec86a83738b85276dd65cb1810102
* :green_heart: [Build #63736](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63736/) succeeded 62ac4ea0978f1317186c547181d50525fb0f9359

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63736/","number":"63736","commit":"62ac4ea0978f1317186c547181d50525fb0f9359"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/64562/","number":"64562","commit":"9785d3d06eeec86a83738b85276dd65cb1810102"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65001/","number":"65001","commit":"528c28465051eb9a4469794d0b50bb3c696a1132"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65421/","number":"65421","commit":"446e94b38660616f2cc66d5019cf03e2f26b06f2"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65572/","number":"65572","commit":"0975904b04fe069eb8e254e7976fa5dbcd224739"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65604/)
* Commit: 28977890ac7c962ae4b5fc55cab1a4f3b1184a1e
* This comment will update when the build is complete

### Failed CI Steps
* [Execute kibana-ciGroup1](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65604/execution/node/372/log)
* [Execute kibana-ciGroup1](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65604/execution/node/438/log)

### History
* :broken_heart: [Build #65572](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65572/) failed 0975904b04fe069eb8e254e7976fa5dbcd224739
* :broken_heart: [Build #65421](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65421/) failed 446e94b38660616f2cc66d5019cf03e2f26b06f2
* :broken_heart: [Build #65001](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65001/) failed 528c28465051eb9a4469794d0b50bb3c696a1132
* :broken_heart: [Build #64562](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/64562/) failed 9785d3d06eeec86a83738b85276dd65cb1810102
* :green_heart: [Build #63736](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63736/) succeeded 62ac4ea0978f1317186c547181d50525fb0f9359

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63736/","number":"63736","commit":"62ac4ea0978f1317186c547181d50525fb0f9359"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/64562/","number":"64562","commit":"9785d3d06eeec86a83738b85276dd65cb1810102"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65001/","number":"65001","commit":"528c28465051eb9a4469794d0b50bb3c696a1132"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65421/","number":"65421","commit":"446e94b38660616f2cc66d5019cf03e2f26b06f2"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65572/","number":"65572","commit":"0975904b04fe069eb8e254e7976fa5dbcd224739"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65604/","number":"65604","commit":"28977890ac7c962ae4b5fc55cab1a4f3b1184a1e"}]}
PIPELINE-->

elasticmachine commented:
Pinging @elastic/es-ui (Team:Elasticsearch UI)

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65860/)
* Commit: 2554f4675500c8612c410de92a6483bbfb305582

### Build metrics
#### @kbn/optimizer bundle module count
| id | [value](https://github.com/elastic/kibana/commit/2554f4675500c8612c410de92a6483bbfb305582) | diff | [baseline](https://github.com/elastic/kibana/commit/69844e45eb8247344ea913d006680deced5d3b34) |
| --- | --- | --- | --- |
| esUiShared | 228 | +2 | 226 |

#### page load bundle size
| id | [value](https://github.com/elastic/kibana/commit/2554f4675500c8612c410de92a6483bbfb305582) | diff | [baseline](https://github.com/elastic/kibana/commit/69844e45eb8247344ea913d006680deced5d3b34) |
| --- | --- | --- | --- |
| esUiShared | 987.4KB | +101.0B | 987.3KB |
| indexManagement | 257.4KB | -8.0B | 257.4KB |
| total | - | +93.0B | - |

### History
* :broken_heart: [Build #65604](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65604/) failed 28977890ac7c962ae4b5fc55cab1a4f3b1184a1e
* :broken_heart: [Build #65572](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65572/) failed 0975904b04fe069eb8e254e7976fa5dbcd224739
* :broken_heart: [Build #65421](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65421/) failed 446e94b38660616f2cc66d5019cf03e2f26b06f2
* :broken_heart: [Build #65001](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65001/) failed 528c28465051eb9a4469794d0b50bb3c696a1132
* :broken_heart: [Build #64562](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/64562/) failed 9785d3d06eeec86a83738b85276dd65cb1810102

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/64562/","number":"64562","commit":"9785d3d06eeec86a83738b85276dd65cb1810102"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65001/","number":"65001","commit":"528c28465051eb9a4469794d0b50bb3c696a1132"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65421/","number":"65421","commit":"446e94b38660616f2cc66d5019cf03e2f26b06f2"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65572/","number":"65572","commit":"0975904b04fe069eb8e254e7976fa5dbcd224739"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65604/","number":"65604","commit":"28977890ac7c962ae4b5fc55cab1a4f3b1184a1e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65860/","number":"65860","commit":"2554f4675500c8612c410de92a6483bbfb305582"}]}
PIPELINE-->

cjcenizal commented on src/plugins/es_ui_shared/public/request/use_request.ts:
```diff
@@ -126,50 +94,44 @@ export const useRequest = <D = any, E = Error>(
     // But it doesn't seem likely that the method will change and body is likely to be a new
     // object even if its shape hasn't changed, so for now we're just watching the path and the query.
-  }, [path, stringifiedQuery]);
+  }, [path, query]);
```
These are only dependencies that I think should have any effect on the hook behavior, at least at this point. If we were to change the refs above to `useCallback`, I'm concerned about introducing unwanted side effects. For example, converting `sendRequestRef` to `useCallback` means defining dependencies that duplicate those defined here (see below).

I'm open to exploring this in separate PRs, but I think for the context of this PR I'd like to avoid introducing any complexity that doesn't have a clear, deliberate purpose (e.g. fixing a bug or defining behavior more clearly). I think the current code makes it clear what will happen if a) dependencies change and b) `sendRequestRef` is called. If other people feel the same way then I think that's enough for me to consider this merge-worthy in its current form.

```ts
  const sendRequest = useCallback(async () => {
    // We don't clear error or data, so it's up to the consumer to decide whether to display the
    // "old" error/data or loading state when a new request is in-flight.
    setIsLoading(true);

    const requestBody = {
      path,
      method,
      query,
      body,
    };

    const response = await sendRequest<D, E>(httpClient, requestBody);
    const { data: serializedResponseData, error: responseError } = response;

    // If an outdated request has resolved, ignore its outdated data.
    if (isOutdatedRequest) {
      return { data: null, error: null };
    }

    setError(responseError);
    // If there's an error, keep the data from the last request in case it's still useful to the user.
    if (!responseError) {
      const responseData = deserializer(serializedResponseData);
      setData(responseData);
    }
    setIsLoading(false);
    setIsInitialRequest(false);

    // If we're on an interval, we need to schedule the next request. This also allows us to reset
    // the interval if the user has manually requested the data, to avoid doubled-up requests.
    scheduleRequestRef.current!();

    return { data: serializedResponseData, error: responseError };
  }, [body, deserializer, httpClient, isOutdatedRequest, method, path, query]);
```

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65981/)
* Commit: 2fe8e19fa82b8a08001fbb23063dea59047147e8
* This comment will update when the build is complete

### History
* :green_heart: [Build #65860](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65860/) succeeded 2554f4675500c8612c410de92a6483bbfb305582
* :broken_heart: [Build #65604](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65604/) failed 28977890ac7c962ae4b5fc55cab1a4f3b1184a1e
* :broken_heart: [Build #65572](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65572/) failed 0975904b04fe069eb8e254e7976fa5dbcd224739
* :broken_heart: [Build #65421](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65421/) failed 446e94b38660616f2cc66d5019cf03e2f26b06f2
* :broken_heart: [Build #65001](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65001/) failed 528c28465051eb9a4469794d0b50bb3c696a1132

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65001/","number":"65001","commit":"528c28465051eb9a4469794d0b50bb3c696a1132"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65421/","number":"65421","commit":"446e94b38660616f2cc66d5019cf03e2f26b06f2"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65572/","number":"65572","commit":"0975904b04fe069eb8e254e7976fa5dbcd224739"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65604/","number":"65604","commit":"28977890ac7c962ae4b5fc55cab1a4f3b1184a1e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65860/","number":"65860","commit":"2554f4675500c8612c410de92a6483bbfb305582"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65981/","number":"65981","commit":"2fe8e19fa82b8a08001fbb23063dea59047147e8"}]}
PIPELINE-->

cjcenizal commented:
CC @walterra @jen-huang. The code here is the same as what you approved in https://github.com/elastic/kibana/pull/59626. I'm tagging you for visibility, but I intend to merge this once @sebelga approves. We have most of the release cycle to detect any bugs introduced into the consuming apps by these changes.

One subtle change that could cause a change in behavior is in the way `sendRequest` returns errors. It changed from this:

```js
return {
  data: null,
  error: e.response ? e.response : e,
};
```

To this:

```js
return {
  data: null,
  error: e.response && e.response.data ? e.response.data : e.body,
};
```

To find and fix any bugs means we'll need to look through Snapshot Restore and Ingest Manager, which depended upon the old behavior.

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66167/)
* Commit: ea6d8e4d16f6d5c5697b7a260f200cdf17e2dad2
* This comment will update when the build is complete

### History
* :broken_heart: [Build #65981](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65981/) failed 2fe8e19fa82b8a08001fbb23063dea59047147e8
* :green_heart: [Build #65860](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65860/) succeeded 2554f4675500c8612c410de92a6483bbfb305582
* :broken_heart: [Build #65604](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65604/) failed 28977890ac7c962ae4b5fc55cab1a4f3b1184a1e
* :broken_heart: [Build #65572](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65572/) failed 0975904b04fe069eb8e254e7976fa5dbcd224739
* :broken_heart: [Build #65421](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65421/) failed 446e94b38660616f2cc66d5019cf03e2f26b06f2

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65421/","number":"65421","commit":"446e94b38660616f2cc66d5019cf03e2f26b06f2"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65572/","number":"65572","commit":"0975904b04fe069eb8e254e7976fa5dbcd224739"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65604/","number":"65604","commit":"28977890ac7c962ae4b5fc55cab1a4f3b1184a1e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65860/","number":"65860","commit":"2554f4675500c8612c410de92a6483bbfb305582"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65981/","number":"65981","commit":"2fe8e19fa82b8a08001fbb23063dea59047147e8"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66167/","number":"66167","commit":"ea6d8e4d16f6d5c5697b7a260f200cdf17e2dad2"}]}
PIPELINE-->

cjcenizal commented on src/plugins/es_ui_shared/public/request/use_request.helpers.tsx:
```diff
@@ -0,0 +1,181 @@
+    });
+  };
+
+  const advanceTime = async (ms: number) => {
```
If we were to set a default here, you'd have to be familiar with the `advanceTime` implementation to know what that default is in order to read the code where it's called and understand what's going on. I find it easier to reason about the code by reading `advanceTime(REQUEST_TIME)` rather than `advanceTime()` and then recalling, "This will default to `REQUEST_TIME` in the implementation".

cjcenizal commented on src/plugins/es_ui_shared/public/request/use_request.test.ts:
```diff
@@ -0,0 +1,293 @@
+        sinon.assert.calledWith(deserializer, getSuccessResponse().data);
+      });
+
+      it('processes data', async () => {
```
The first test verifies that the `deserializer` is called with whatever a successful request resolves with. The second test verifies that whatever the `deserializer` returns is returned as `data`. So they're both related, but complementary -- one tests what goes into `deserializer` and the the other tests what comes out of `sendRequest`.

cjcenizal commented on src/plugins/es_ui_shared/public/request/use_request.helpers.tsx:
```diff
@@ -0,0 +1,181 @@
+  const sendRequestSpy = sinon.stub();
+
+  const setupUseRequest = (config: SendRequestConfig) => {
+    const httpClient = {
```
I originally moved it inside for clarity and readability, since the performance hit seems negligible. By placing it inside `setupUseRequest` it makes it clear that this is the only place it's consumed.

cjcenizal commented on src/plugins/es_ui_shared/public/request/use_request.helpers.tsx:
```diff
@@ -0,0 +1,181 @@
+        requestConfig
+      );
+
+      hookResult.isInitialRequest = isInitialRequest;
```
👍  Thanks for the suggestion! I appreciate the suggestion and I've decided that I prefer the simplicity of the current solution. There are fewer moving parts.

cjcenizal commented on src/plugins/es_ui_shared/public/request/send_request.helpers.ts:
```diff
@@ -0,0 +1,83 @@
+/*
```
That's a good point! I can definitely see people being confused by the name. I like colocating tests with the files they test, so I've decided to name `helpers` to `test.helpers` to disambiguate their role.

cjcenizal commented:
@sebelga Thanks for catching that bug! I accidentally introduced that in 2fe8e19, so I reverted the commit and added a comment to explain that line's role.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66515/)
* Commit: b83fc418884441ebdfe2b8f5607a2fda118d345e

### Build metrics
#### @kbn/optimizer bundle module count
| id | [value](https://github.com/elastic/kibana/commit/b83fc418884441ebdfe2b8f5607a2fda118d345e) | diff | [baseline](https://github.com/elastic/kibana/commit/89f1ac49870d06624eca635b82c83bce4f0545df) |
| --- | --- | --- | --- |
| esUiShared | 228 | +2 | 226 |

#### page load bundle size
| id | [value](https://github.com/elastic/kibana/commit/b83fc418884441ebdfe2b8f5607a2fda118d345e) | diff | [baseline](https://github.com/elastic/kibana/commit/89f1ac49870d06624eca635b82c83bce4f0545df) |
| --- | --- | --- | --- |
| esUiShared | 987.4KB | +101.0B | 987.3KB |
| indexManagement | 257.4KB | -8.0B | 257.4KB |
| total | - | +93.0B | - |

### History
* :broken_heart: [Build #66167](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66167/) failed ea6d8e4d16f6d5c5697b7a260f200cdf17e2dad2
* :broken_heart: [Build #65981](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65981/) failed 2fe8e19fa82b8a08001fbb23063dea59047147e8
* :green_heart: [Build #65860](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65860/) succeeded 2554f4675500c8612c410de92a6483bbfb305582
* :broken_heart: [Build #65604](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65604/) failed 28977890ac7c962ae4b5fc55cab1a4f3b1184a1e
* :broken_heart: [Build #65572](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65572/) failed 0975904b04fe069eb8e254e7976fa5dbcd224739

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65572/","number":"65572","commit":"0975904b04fe069eb8e254e7976fa5dbcd224739"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65604/","number":"65604","commit":"28977890ac7c962ae4b5fc55cab1a4f3b1184a1e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65860/","number":"65860","commit":"2554f4675500c8612c410de92a6483bbfb305582"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65981/","number":"65981","commit":"2fe8e19fa82b8a08001fbb23063dea59047147e8"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66167/","number":"66167","commit":"ea6d8e4d16f6d5c5697b7a260f200cdf17e2dad2"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66515/","number":"66515","commit":"b83fc418884441ebdfe2b8f5607a2fda118d345e"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66524/)
* Commit: ded10edbfc0b3043a614312be98ad3d3eeccf830

### Build metrics
#### @kbn/optimizer bundle module count
| id | [value](https://github.com/elastic/kibana/commit/ded10edbfc0b3043a614312be98ad3d3eeccf830) | diff | [baseline](https://github.com/elastic/kibana/commit/89f1ac49870d06624eca635b82c83bce4f0545df) |
| --- | --- | --- | --- |
| esUiShared | 228 | +2 | 226 |

#### page load bundle size
| id | [value](https://github.com/elastic/kibana/commit/ded10edbfc0b3043a614312be98ad3d3eeccf830) | diff | [baseline](https://github.com/elastic/kibana/commit/89f1ac49870d06624eca635b82c83bce4f0545df) |
| --- | --- | --- | --- |
| esUiShared | 987.4KB | +101.0B | 987.3KB |
| indexManagement | 257.4KB | -8.0B | 257.4KB |
| total | - | +93.0B | - |

### History
* :green_heart: [Build #66515](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66515/) succeeded b83fc418884441ebdfe2b8f5607a2fda118d345e
* :broken_heart: [Build #66167](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66167/) failed ea6d8e4d16f6d5c5697b7a260f200cdf17e2dad2
* :broken_heart: [Build #65981](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65981/) failed 2fe8e19fa82b8a08001fbb23063dea59047147e8
* :green_heart: [Build #65860](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65860/) succeeded 2554f4675500c8612c410de92a6483bbfb305582
* :broken_heart: [Build #65604](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65604/) failed 28977890ac7c962ae4b5fc55cab1a4f3b1184a1e

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65604/","number":"65604","commit":"28977890ac7c962ae4b5fc55cab1a4f3b1184a1e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65860/","number":"65860","commit":"2554f4675500c8612c410de92a6483bbfb305582"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65981/","number":"65981","commit":"2fe8e19fa82b8a08001fbb23063dea59047147e8"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66167/","number":"66167","commit":"ea6d8e4d16f6d5c5697b7a260f200cdf17e2dad2"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66515/","number":"66515","commit":"b83fc418884441ebdfe2b8f5607a2fda118d345e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66524/","number":"66524","commit":"ded10edbfc0b3043a614312be98ad3d3eeccf830"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66807/)
* Commit: c4ab04ddd9670e4226b19a9d5cdee814848e11ca
* This comment will update when the build is complete

### Failed CI Steps
* [Lint: eslint](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66807/execution/node/348/log)

### History
* :green_heart: [Build #66524](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66524/) succeeded ded10edbfc0b3043a614312be98ad3d3eeccf830
* :green_heart: [Build #66515](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66515/) succeeded b83fc418884441ebdfe2b8f5607a2fda118d345e
* :broken_heart: [Build #66167](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66167/) failed ea6d8e4d16f6d5c5697b7a260f200cdf17e2dad2
* :broken_heart: [Build #65981](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65981/) failed 2fe8e19fa82b8a08001fbb23063dea59047147e8
* :green_heart: [Build #65860](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65860/) succeeded 2554f4675500c8612c410de92a6483bbfb305582

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65860/","number":"65860","commit":"2554f4675500c8612c410de92a6483bbfb305582"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65981/","number":"65981","commit":"2fe8e19fa82b8a08001fbb23063dea59047147e8"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66167/","number":"66167","commit":"ea6d8e4d16f6d5c5697b7a260f200cdf17e2dad2"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66515/","number":"66515","commit":"b83fc418884441ebdfe2b8f5607a2fda118d345e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66524/","number":"66524","commit":"ded10edbfc0b3043a614312be98ad3d3eeccf830"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66807/","number":"66807","commit":"c4ab04ddd9670e4226b19a9d5cdee814848e11ca"}]}
PIPELINE-->

cjcenizal commented on src/plugins/es_ui_shared/public/request/use_request.ts:
```diff
@@ -0,0 +1,167 @@
+    if (!isLoading) {
+      scheduleRequest();
+    }
+  }, [isLoading, scheduleRequest]);
```
Note that here we're using `isLoading` to signal the need to reschedule a request, and not `totalRequests` as in https://github.com/cjcenizal/kibana/pull/25/files#diff-a60f7c6c34d3d037970d34856d1b3da2R142. This is so that we don't interfere with the polling rhythm whenever `sendRequest` dependencies change.

sebelga replied:
I think it is a little bit less explicit than with my `totalRequest` count (I personally understand better "when the request count increments, reschedule a request", as "isLoading" _seems_ like it could change independently.) but I am OK to go with your suggestion.

But your comment is not right, the reschedule **will** happen whenever `scheduleRequest` changes, thus whenever its dependencies change.

Also, you are missing a 

```js
if (!isMounted.current) {
  return;
}
```

on top, as we don't want to schedule a request when the component mounts, but when the initial request finished.

cjcenizal replied:
> But your comment is not right, the reschedule will happen whenever scheduleRequest changes, thus whenever its dependencies change.

Could you explain this a bit? From the way I read this code, it looks like the effect is called whenever `scheduleRequest` (and therefore `sendRequest`) changes, but the `isLoading` condition prevents `scheduleRequest` from being called each time.

> if (!isMounted.current) {
>   return;
> }

`isLoading` is true when the component mounts, so the request isn't being initially scheduled. I'll add a comment to clarify this.

cjcenizal replied:
For context, I had to switch away from using `totalRequests` because here we're incrementing it before the request completes to get the request ID, instead of after like you originally had. This meant that if we used it to schedule the next request, then the next request would be scheduled before the first one completes.

sebelga replied:
> isLoading is true when the component mounts

This is the problem I think. It should be `false` (as I was expecting it) and it is set to true when `sendRequest()` is called.

> Could you explain this a bit? 

Nevermind I will understand it. For me adding so much comment makes it harder to follow and it seems that there is something "special" the code, but that's ok.
I was saying that the ´useEffect´ **will be** triggered whenever any of the deps changes (which is simply how useEffect works). And of course, the `if` condition prevents the reschedule. But I don't need a comment to explain that.

My mental model was

```js
const [isLoading, setIsLoading] = useState(false); // nothing happened just yet

...

useEffect(() => {
  if (!isMounted.current) {
    return; // clearly indicate to not schedule anything on mount
  }

  if (!isLoading) {
    scheduleRequest();
  }

}, [isLoading, scheduleRequest]);
```
With the above code, I don't need any comment as it is self explanatory.

> For context, I had to switch away from using totalRequests because here we're incrementing it before the request completes to get the request ID, instead of after like you originally suggested. This meant that if we used it to schedule the next request, then the next request would be scheduled before the first one completes.

I am not following but that's OK I guess 😄

cjcenizal commented:
@sebelga I've converted all of `sendRequest`'s internals to hooks, with a couple key differences from your original suggestion. I added some extra tests to define the behavior I think we want. Can you please take another look?

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71148/)
* Commit: 6ff61248c848ff924f9e89358337b0c74c465e47
* This comment will update when the build is complete

### Failed CI Steps
* [Lint: eslint](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71148/execution/node/346/log)

### History
* :broken_heart: [Build #66807](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66807/) failed c4ab04ddd9670e4226b19a9d5cdee814848e11ca
* :green_heart: [Build #66524](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66524/) succeeded ded10edbfc0b3043a614312be98ad3d3eeccf830
* :green_heart: [Build #66515](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66515/) succeeded b83fc418884441ebdfe2b8f5607a2fda118d345e
* :broken_heart: [Build #66167](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66167/) failed ea6d8e4d16f6d5c5697b7a260f200cdf17e2dad2
* :broken_heart: [Build #65981](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65981/) failed 2fe8e19fa82b8a08001fbb23063dea59047147e8

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65981/","number":"65981","commit":"2fe8e19fa82b8a08001fbb23063dea59047147e8"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66167/","number":"66167","commit":"ea6d8e4d16f6d5c5697b7a260f200cdf17e2dad2"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66515/","number":"66515","commit":"b83fc418884441ebdfe2b8f5607a2fda118d345e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66524/","number":"66524","commit":"ded10edbfc0b3043a614312be98ad3d3eeccf830"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66807/","number":"66807","commit":"c4ab04ddd9670e4226b19a9d5cdee814848e11ca"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71148/","number":"71148","commit":"6ff61248c848ff924f9e89358337b0c74c465e47"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71155/)
* Commit: 40820aa3adc2767a326f09e1842383dca637e52f
* This comment will update when the build is complete

### History
* :broken_heart: [Build #71148](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71148/) failed 6ff61248c848ff924f9e89358337b0c74c465e47
* :broken_heart: [Build #66807](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66807/) failed c4ab04ddd9670e4226b19a9d5cdee814848e11ca
* :green_heart: [Build #66524](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66524/) succeeded ded10edbfc0b3043a614312be98ad3d3eeccf830
* :green_heart: [Build #66515](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66515/) succeeded b83fc418884441ebdfe2b8f5607a2fda118d345e
* :broken_heart: [Build #66167](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66167/) failed ea6d8e4d16f6d5c5697b7a260f200cdf17e2dad2

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66167/","number":"66167","commit":"ea6d8e4d16f6d5c5697b7a260f200cdf17e2dad2"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66515/","number":"66515","commit":"b83fc418884441ebdfe2b8f5607a2fda118d345e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66524/","number":"66524","commit":"ded10edbfc0b3043a614312be98ad3d3eeccf830"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66807/","number":"66807","commit":"c4ab04ddd9670e4226b19a9d5cdee814848e11ca"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71148/","number":"71148","commit":"6ff61248c848ff924f9e89358337b0c74c465e47"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71155/","number":"71155","commit":"40820aa3adc2767a326f09e1842383dca637e52f"}]}
PIPELINE-->

cjcenizal commented:
@elasticmachine merge upstream

cjcenizal commented on src/plugins/es_ui_shared/public/request/use_request.ts:
```diff
@@ -0,0 +1,162 @@
+    // Setting isLoading to false also acts as a signal for scheduling the next poll request.
+    setIsLoading(false);
+
+    return { data: serializedResponseData, error: responseError };
```
I think it's probably an antipattern that we're returning values from `sendRequest`. Seems like the consumer should be depending on the values returned by the hook instead. Once we merge this I'll create an issue to check for any consumers that depend on this return value and refactor them so we can try to remove this.

sebelga replied:
👍  This is something I noticed since the first iteration. But for some reason, we added the `"// Gives the user the ability to manually request data"` comment and exposed the handler.

cjcenizal replied:
I think we do want to retain `sendRequest` as part of the `useRequest` hook's return value, because we need it for "Reload" button behavior. But I don't think calling `sendRequest` should return a promise that resolves to `data` and `error`, since that information is already available off the `useRequest` hook's return value.

sebelga replied:
Ah ok I see, then maybe the API should be clearer and expose a `refresh()` handler that indeed does not return `data` nor `error`.

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71163/)
* Commit: fc2f91eb4af91dc3a7a2a587158a388ce4f85ddf
* This comment will update when the build is complete

### History
* :broken_heart: [Build #71155](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71155/) failed 40820aa3adc2767a326f09e1842383dca637e52f
* :broken_heart: [Build #71148](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71148/) failed 6ff61248c848ff924f9e89358337b0c74c465e47
* :broken_heart: [Build #66807](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66807/) failed c4ab04ddd9670e4226b19a9d5cdee814848e11ca
* :green_heart: [Build #66524](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66524/) succeeded ded10edbfc0b3043a614312be98ad3d3eeccf830
* :green_heart: [Build #66515](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66515/) succeeded b83fc418884441ebdfe2b8f5607a2fda118d345e

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66515/","number":"66515","commit":"b83fc418884441ebdfe2b8f5607a2fda118d345e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66524/","number":"66524","commit":"ded10edbfc0b3043a614312be98ad3d3eeccf830"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66807/","number":"66807","commit":"c4ab04ddd9670e4226b19a9d5cdee814848e11ca"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71148/","number":"71148","commit":"6ff61248c848ff924f9e89358337b0c74c465e47"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71155/","number":"71155","commit":"40820aa3adc2767a326f09e1842383dca637e52f"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71163/","number":"71163","commit":"fc2f91eb4af91dc3a7a2a587158a388ce4f85ddf"}]}
PIPELINE-->

sebelga commented on src/plugins/es_ui_shared/public/request/use_request.ts:
```diff
@@ -0,0 +1,162 @@
+    clearPollInterval();
+
+    // Pull off request ID and increment request count.
+    const requestId = requestCountRef.current++;
```
nit: here it would be better to have 

```js
const requestId = ++requestCountRef.current;
```

and not have the `requestCountRef.current - 1;` below (the `-1` part)

sebelga commented on src/plugins/es_ui_shared/public/request/use_request.ts:
```diff
@@ -0,0 +1,162 @@
+    // If there's a scheduled poll request, this new one will supersede it.
+    clearPollInterval();
+
+    // Schedule next poll request.
```
nit: same here about unnecessary comment

sebelga commented on src/plugins/es_ui_shared/public/request/use_request.ts:
```diff
@@ -0,0 +1,162 @@
+    }
+  }, [pollIntervalMs, sendRequest, clearPollInterval]);
+
+  // Send the request on component mount and whenever the dependencies of sendRequest() change.
```
nit: same here about the unnecessary comment (here you are basically describing the `useEffect` lifecycle.

cjcenizal replied:
I've removed the other comments you pointed out. I think this is the one comment I'm going to keep, just to really emphasize "the dependencies of sendRequest()" bit. It seems super obvious after staring at this code for hours, but I think the comment is more for someone who hasn't already spent a ton of time looking at the code. :)

sebelga commented on src/plugins/es_ui_shared/public/request/use_request.ts:
```diff
@@ -0,0 +1,162 @@
+    // data, to avoid doubled-up requests.
+    clearPollInterval();
+
+    // Pull off request ID and increment request count.
```
nit: I am not sure it is necessary to comment on obvious code behavior 😊  As per our guidelines "_Don't use comments to restate trivial things._" (https://github.com/elastic/kibana/blob/master/STYLEGUIDE.md#use-slashes-for-comments)

sebelga commented on .eslintrc.js:
```diff
@@ -94,12 +94,6 @@ module.exports = {
         'jsx-a11y/no-onchange': 'off',
       },
     },
-    {
```
Great! Thanks for removing it here and adding the comments in the few places we still have the issue 👍

sebelga reviewed (approved):
Awesome work @cjcenizal ! We almost got it! 😊 I left some not about unnecessary comments that "_restate trivial things_". For the hook I left a comment about a missing mount check that should be added. I will approve but it'd be good to add it before merging. Thanks!

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71242/)
* Commit: b3414b7d4cd4eec296483ac46bbe5a75bd4eecff
* This comment will update when the build is complete

### History
* :broken_heart: [Build #71163](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71163/) failed fc2f91eb4af91dc3a7a2a587158a388ce4f85ddf
* :broken_heart: [Build #71155](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71155/) failed 40820aa3adc2767a326f09e1842383dca637e52f
* :broken_heart: [Build #71148](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71148/) failed 6ff61248c848ff924f9e89358337b0c74c465e47
* :broken_heart: [Build #66807](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66807/) failed c4ab04ddd9670e4226b19a9d5cdee814848e11ca
* :green_heart: [Build #66524](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66524/) succeeded ded10edbfc0b3043a614312be98ad3d3eeccf830

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66524/","number":"66524","commit":"ded10edbfc0b3043a614312be98ad3d3eeccf830"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66807/","number":"66807","commit":"c4ab04ddd9670e4226b19a9d5cdee814848e11ca"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71148/","number":"71148","commit":"6ff61248c848ff924f9e89358337b0c74c465e47"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71155/","number":"71155","commit":"40820aa3adc2767a326f09e1842383dca637e52f"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71163/","number":"71163","commit":"fc2f91eb4af91dc3a7a2a587158a388ce4f85ddf"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71242/","number":"71242","commit":"b3414b7d4cd4eec296483ac46bbe5a75bd4eecff"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71378/)
* Commit: 0e99888dd017f8553406c3b4812d925e07492439

### Build metrics
#### @kbn/optimizer bundle module count
| id | [value](https://github.com/elastic/kibana/commit/0e99888dd017f8553406c3b4812d925e07492439) | diff | [baseline](https://github.com/elastic/kibana/commit/1e20cd4cf44fe1395cc99277a6ac940adb236f49) |
| --- | --- | --- | --- |
| esUiShared | 229 | +2 | 227 |

#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/0e99888dd017f8553406c3b4812d925e07492439) | diff | [baseline](https://github.com/elastic/kibana/commit/1e20cd4cf44fe1395cc99277a6ac940adb236f49) |
| --- | --- | --- | --- |
| watcher | 602.5KB | +602.0B | 602.0KB |

#### page load bundle size
| id | [value](https://github.com/elastic/kibana/commit/0e99888dd017f8553406c3b4812d925e07492439) | diff | [baseline](https://github.com/elastic/kibana/commit/1e20cd4cf44fe1395cc99277a6ac940adb236f49) |
| --- | --- | --- | --- |
| esUiShared | 991.0KB | +462.0B | 990.6KB |
| indexManagement | 248.1KB | -8.0B | 248.1KB |
| total |  | +454.0B |  |

### History
* :broken_heart: [Build #71242](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71242/) failed b3414b7d4cd4eec296483ac46bbe5a75bd4eecff
* :broken_heart: [Build #71163](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71163/) failed fc2f91eb4af91dc3a7a2a587158a388ce4f85ddf
* :broken_heart: [Build #71155](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71155/) failed 40820aa3adc2767a326f09e1842383dca637e52f
* :broken_heart: [Build #71148](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71148/) failed 6ff61248c848ff924f9e89358337b0c74c465e47
* :broken_heart: [Build #66807](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66807/) failed c4ab04ddd9670e4226b19a9d5cdee814848e11ca

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66807/","number":"66807","commit":"c4ab04ddd9670e4226b19a9d5cdee814848e11ca"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71148/","number":"71148","commit":"6ff61248c848ff924f9e89358337b0c74c465e47"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71155/","number":"71155","commit":"40820aa3adc2767a326f09e1842383dca637e52f"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71163/","number":"71163","commit":"fc2f91eb4af91dc3a7a2a587158a388ce4f85ddf"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71242/","number":"71242","commit":"b3414b7d4cd4eec296483ac46bbe5a75bd4eecff"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71378/","number":"71378","commit":"0e99888dd017f8553406c3b4812d925e07492439"}]}
PIPELINE-->

cjcenizal commented:
@elastic/ingest-management @elastic/ml-ui Sorry for not pinging you before merging this! Could you please test out your apps which consume `useRequest` and verify that they still function correctly? I'd look extra carefully at your error-handling. If you specify the `body`, `query`, `deserializer`, or `pollIntervalMs` arguments, I'd also test that logic in particular.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72514/)
* Commit: 0e99888dd017f8553406c3b4812d925e07492439

### Build metrics
#### @kbn/optimizer bundle module count
| id | [value](https://github.com/elastic/kibana/commit/0e99888dd017f8553406c3b4812d925e07492439) | diff | [baseline](https://github.com/elastic/kibana/commit/1e20cd4cf44fe1395cc99277a6ac940adb236f49) |
| --- | --- | --- | --- |
| esUiShared | 229 | +2 | 227 |

#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/0e99888dd017f8553406c3b4812d925e07492439) | diff | [baseline](https://github.com/elastic/kibana/commit/1e20cd4cf44fe1395cc99277a6ac940adb236f49) |
| --- | --- | --- | --- |
| watcher | 602.5KB | +602.0B | 602.0KB |

#### page load bundle size
| id | [value](https://github.com/elastic/kibana/commit/0e99888dd017f8553406c3b4812d925e07492439) | diff | [baseline](https://github.com/elastic/kibana/commit/1e20cd4cf44fe1395cc99277a6ac940adb236f49) |
| --- | --- | --- | --- |
| esUiShared | 991.0KB | +462.0B | 990.6KB |
| indexManagement | 248.1KB | -8.0B | 248.1KB |
| total |  | +454.0B |  |

### History
* :green_heart: [Build #71378](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71378/) succeeded 0e99888dd017f8553406c3b4812d925e07492439
* :broken_heart: [Build #71242](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71242/) failed b3414b7d4cd4eec296483ac46bbe5a75bd4eecff
* :broken_heart: [Build #71163](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71163/) failed fc2f91eb4af91dc3a7a2a587158a388ce4f85ddf
* :broken_heart: [Build #71155](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71155/) failed 40820aa3adc2767a326f09e1842383dca637e52f
* :broken_heart: [Build #71148](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71148/) failed 6ff61248c848ff924f9e89358337b0c74c465e47

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71148/","number":"71148","commit":"6ff61248c848ff924f9e89358337b0c74c465e47"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71155/","number":"71155","commit":"40820aa3adc2767a326f09e1842383dca637e52f"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71163/","number":"71163","commit":"fc2f91eb4af91dc3a7a2a587158a388ce4f85ddf"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71242/","number":"71242","commit":"b3414b7d4cd4eec296483ac46bbe5a75bd4eecff"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71378/","number":"71378","commit":"0e99888dd017f8553406c3b4812d925e07492439"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72514/","number":"72514","commit":"0e99888dd017f8553406c3b4812d925e07492439"}]}
PIPELINE-->
